rpc/transport/transport.go

```diff
@@ -147,7 +153,17 @@ func (s *transport) NewMessage(ctx context.Context) (_ rpccp.Message, send func(
 		return err
 	}
 
-	return rmsg, send, func() { msg.Reset(nil) }, nil
+	release = func() {
+		if alreadyReleased {
```

Comment:
This won't protect us from concurrent calls to `release()`. Is that a problem?

Go 1.19's `atomic.Bool.Swap(true)` might be a good fit, here.

Reply:
It's not a problem for the places it's actually used (which afaik is just the tests). It feels wrong to introduce atomics for something production code doesn't actually rely on...

Reply:
SGTM.  Resolving.